test/prober/receiver.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 The Knative Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package prober
+
+import (
+	"fmt"
+
+	"github.com/wavesoftware/go-ensure"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"knative.dev/eventing/test/lib"
+	"knative.dev/pkg/test"
+)
+
+var (
+	receiverName           = "wathola-receiver"
+	receiverNodePort int32 = -1
+)
+
+func (p *prober) deployReceiver() {
+	p.deployReceiverPod()
+	p.deployReceiverService()
+}
+
+func (p *prober) removeReceiver() {
+	p.removeReceiverService()
+	p.removeReceiverPod()
+}
+
+func (p *prober) deployReceiverPod() {
+	p.log.Infof("Deploy of receiver pod: %v", receiverName)
+	pod := &corev1.Pod{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      receiverName,
+			Namespace: p.config.Namespace,
+			Labels: map[string]string{
+				"app": receiverName,
+			},
+		},
+		Spec: corev1.PodSpec{
+			Volumes: []corev1.Volume{
+				{
+					Name: configName,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{Name: configName},
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "receiver",
+					Image: fmt.Sprintf("quay.io/cardil/wathola-receiver:%v", Version),
```

Comment:
the normal pattern in eventing is if a tool is required to be built, it is build and the resolved image is passed into the tool. it is a red flag to pull something from quay.

Reply:
Also, that concern is already raised in this thread https://github.com/knative/eventing/pull/2388/files/3746ec01ec63fe65d420a5fe4105fddfb17d9e7d#r406279058

Please see my latest comments in there.

Reply:
This tool was incorporated in this PR, and no longer is pulled from quay.io